From 09282853e82a1a881e7e5b061030063f48df2b95 Mon Sep 17 00:00:00 2001
From: Michael Niedermayer <michael@niedermayer.cc>
Date: Tue, 6 Oct 2026 06:53:33 +0200
Subject: [PATCH] avformat/rtmppkt: Terminate truncated AMF strings at the
 bytes read

Fixes: use of uninitialized memory
Fixes: Fid2Ij03ilhP
Fixes: AISLE-2026-0111-00158
Use of uninitialized memory Replicated through UnModified FFmpeg with Valgrind
Found-by: Joshua Rogers <joshua.rogers@aisle.com>
---
 libavformat/rtmppkt.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libavformat/rtmppkt.c b/libavformat/rtmppkt.c
index ee687d479156..98fb98a9bcce 100644
--- a/libavformat/rtmppkt.c
+++ b/libavformat/rtmppkt.c
@@ -551,7 +551,7 @@ static int amf_get_field_value2(GetByteContext *gb,
                     return -1;
                 if (dst_size < len + 1)
                     len = dst_size - 1;
-                bytestream2_get_buffer(gb, dst, len);
+                len = bytestream2_get_buffer(gb, dst, len);
                 dst[len] = 0;
                 break;
             default:
